SnapWar/ViewControllers/SWAuthViewController.swift

```diff
@@ -28,21 +29,66 @@ class SWAuthViewController: UIViewController {
         formContainer.isLayoutMarginsRelativeArrangement = true
         formContainer.layoutMargins = UIEdgeInsets(top: 0, left: horizontalMargin, bottom: 0, right: horizontalMargin)
         
-        let keyboardTypes = [
-            UIKeyboardType.alphabet,
-            UIKeyboardType.numberPad,
-            UIKeyboardType.emailAddress
-        ]
-        for keyboardType in keyboardTypes {
-            let textField = SWTextField()
-            textField.keyboardType = keyboardType
-            textField.placeholder = "type here"
+        usernameField.placeholder = "username"
+        usernameField.delegate = self
+        usernameField.tag = 0
+        formContainer.addArrangedSubview(usernameField)
 
-            formContainer.addArrangedSubview(textField)
-        }
-        
-        let passwordField = SWPasswordTextField()
         passwordField.placeholder = "password"
+        passwordField.delegate = self
+        passwordField.tag = 1
         formContainer.addArrangedSubview(passwordField)
+        
+        loginButton.setTitle("Login", for: .normal)
+        loginButton.backgroundColor = SWColorType.primary
+        loginButton.addTarget(self, action: #selector(handleSubmit), for: .touchUpInside)
+        setLoginButtonState()
+        formContainer.addArrangedSubview(loginButton)
+    }
+    
+    func textFieldDidEndEditing(_ textField: UITextField) {
+        setLoginButtonState()
+    }
+    
+    func textFieldShouldReturn(_ textField: UITextField) -> Bool {
+        let nextTag = textField.tag + 1
+        
+        if let nextResponder = textField.superview?.viewWithTag(nextTag) {
+            nextResponder.becomeFirstResponder()
+        } else {
+            textField.resignFirstResponder()
```

Comment:
should we invoke the handle submit operation here if the text field here is for password?